src/urllib3/_collections.py

```diff
@@ -28,10 +55,38 @@ def __exit__(self, exc_type, exc_value, traceback):
 __all__ = ["RecentlyUsedContainer", "HTTPHeaderDict"]
 
 
-_Null = object()
-
-
-class RecentlyUsedContainer(MutableMapping):
+_KT = TypeVar("_KT")
+_VT = TypeVar("_VT")
+
+ValidHttpHeaderSource = Union[
```

Comment:
nit: casing as `HTTPHeader` instead of `HttpHeader` matches the casing elsewhere (ie `HTTPHeaderDict`)